feat: modernized docker-compose.yml

Leopere opened the pull request:
# Proposed Changes
- Modernized to the latest specs of Compose.
- set casvisor service to only expose the frontend port and not the backend port on all of the host interfaces.
- removed the unmentioned in the documentation port 3000 on the Casvisor port
- Modernized the restart and deploy strategy to optimize for uptime.
- Switched Casvisor service to the official image produced by Casvisor and pushed to the docker hub registry.
- Unpublished MySQL from all interfaces and is now exposed only internally to this docker-compose stack.
- Created an explicitly defined network that is technically superfluous.
- Staggered deployment so that Casvisor does not immediately start but the 30-second delay is technically arbitrary and can't be predicted as to how long MySQL will take to launch.
- unpublished Guacd from the 0.0.0.0 interface.  Meaning it now must be accessed entirely with docker's internal DNS via the separate default network.  This network could optionally have encryption turned on for those with tin foil hats.
- added a rollback function so that if an update fails to launch the container will automatically fall back to the last known good.

Leopere committed:
Modernized docker-compose.yml

# proposed changes
- Modernized to the latest specs of Compose.
- set casvisor service to only expose the frontend port and not the backend port on all of the host interfaces.
- removed the unmentioned in the documentation port 3000 on the Casvisor port
- Modernized the restart and deploy strategy to optimize for uptime.
- Switched Casvisor service to the official image produced by Casvisor and pushed to the docker hub registry.
- Unpublished MySQL from all interfaces and is now exposed only internally to this docker-compose stack.
- Created an explicitly defined network that is technically superfluous.
- Staggered deployment so that Casvisor does not immediately start but the 30-second delay is technically arbitrary and can't be predicted as to how long MySQL will take to launch.
- unpublished Guacd from the 0.0.0.0 interface.  Meaning it now must be accessed entirely with docker's internal DNS via the separate default network.  This network could optionally have encryption turned on for those with tin foil hats.
- added a rollback function so that if an update fails to launch the container will automatically fall back to the last known good.

leo220yuyaodog commented on docker-compose.yml:
```diff
@@ -1,37 +1,57 @@
     ports:
-      - "19000:19000"
-      - "3000:3000"
+      - "16001:16001"
```
use 19000 port

leo220yuyaodog commented on docker-compose.yml:
```diff
@@ -1,37 +1,57 @@
+      update_config:
+        order: start-first
+        failure_action: rollback
+        delay: 30s # Ensures casvisor starts 30 seconds after db
```
I wonder if 30s is too long, can it be 15s?

Leopere committed:
fix: Resove suggestions for PR #94

switched exposed port to 19000 as per suggested and shortened deploy delay to 15 seconds.
